modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -85,5 +85,31 @@ def report msg, response, error_only
       logger.error "Dnscmd failed:\n" + (response.is_a?(Array) ? response.join("\n") : "Response was not an array! #{response}")
       raise Proxy::Dns::Error.new("Unknown error while processing '#{msg}'")
     end
+
+    def match_zone(fqdn)
+      dns_zones = ::Proxy::Dns::Plugin.settings.dns_zones
+      found_match = false
+      wight = 0
+      matched_zone = fqdn.sub(/[^.]+./,'')
+
+      if dns_zones.is_a?(Array)
```

Comment:
Either need to complain on startup that dns_zones parameter isn't an array (the idea is to report config errors on startup, as it makes troubleshooting easier), or handle this parameter being a string.